---------- normal exceptions --------------------
throw new Exception():                        OK
java.lang.Exception
   at extest.main (extest.java:20)

throw new Exception() (from subroutines):     OK
java.lang.Exception
   at extest.sub4 (extest.java)
   at extest.sub3 (extest.java:387)
   at extest.sub2 (extest.java:383)
   at extest.sub (extest.java:379)
   at extest.main (extest.java:36)

NullPointerException:                         OK
java.lang.NullPointerException
   at extest.main (extest.java:46)


---------- test soft inline exceptions ----------
/* thrown twice to check the inline jump code */
ArithmeticException (only w/ -softnull):      OK
java.lang.ArithmeticException: / by zero
   at extest.main (extest.java:61)

ArithmeticException (only w/ -softnull):      OK
java.lang.ArithmeticException: / by zero
   at extest.main (extest.java:77)

ArrayIndexOutOfBoundsException:               OK
java.lang.ArrayIndexOutOfBoundsException: -889275714
   at extest.main (extest.java:95)

ArrayIndexOutOfBoundsException:               OK
java.lang.ArrayIndexOutOfBoundsException: -1161901314
   at extest.main (extest.java:112)

ArrayStoreException:                          OK
java.lang.ArrayStoreException
   at extest.main (extest.java:132)

ArrayStoreException:                          OK
java.lang.ArrayStoreException
   at extest.main (extest.java:143)

ClassCastException:                           OK
java.lang.ClassCastException
   at extest.main (extest.java:154)

ClassCastException:                           OK
java.lang.ClassCastException
   at extest.main (extest.java:165)

NegativeArraySizeException (newarray):        OK
java.lang.NegativeArraySizeException
   at extest.main (extest.java:175)

NegativeArraySizeException (newarray):        OK
java.lang.NegativeArraySizeException
   at extest.main (extest.java:184)

NegativeArraySizeException (multianewarray):  OK
java.lang.NegativeArraySizeException
   at extest.main (extest.java:193)

NullPointerException (only w/ -softnull):     OK
java.lang.NullPointerException
   at extest.main (extest.java:204)

NullPointerException (only w/ -softnull):     OK
java.lang.NullPointerException
   at extest.main (extest.java:214)

OutOfMemoryError:                             OK
java.lang.OutOfMemoryError
   at extest.main (extest.java:224)

OutOfMemoryError:                             OK
java.lang.OutOfMemoryError
   at extest.main (extest.java:234)

OutOfMemoryError (multianewarray):            OK
java.lang.OutOfMemoryError
   at extest.main (extest.java:244)


---------- exceptions in leaf functions ---------
ArithmeticException:                          OK
java.lang.ArithmeticException: / by zero
   at extest.aesub (extest.java:395)
   at extest.main (extest.java:258)

ArrayIndexOutOfBoundsException:               OK
java.lang.ArrayIndexOutOfBoundsException: -559038737
   at extest.aioobesub (extest.java:399)
   at extest.main (extest.java:267)

ClassCastException:                           OK
java.lang.ClassCastException
   at extest.ccesub (extest.java:403)
   at extest.main (extest.java:284)

NullPointerException:                         OK
java.lang.NullPointerException
   at extest.npesub (extest.java:407)
   at extest.main (extest.java:293)


---------- exception related things -------------
load/link an exception class in asmpart:      OK
java.lang.Exception
   at extest.main (extest.java:306)


---------- native stub exceptions ---------------
NullPointerException in <clinit>:             OK
java.lang.ExceptionInInitializerError
   at extest.main (extest.java:324)
Caused by: java.lang.NullPointerException
   at extest_clinit.<clinit> (extest_clinit.java:4)
   at extest.main (extest.java:324)

UnsatisfiedLinkError:                         OK
java.lang.UnsatisfiedLinkError: nsub
   at extest.nsub (Native Method)
   at extest.main (extest.java:336)

NullPointerException (native):                OK
java.lang.NullPointerException
   at java.lang.VMSystem.arraycopy (Native Method)
   at java.lang.System.arraycopy (System.java:234)
   at extest.main (extest.java:345)


---------- special exceptions -------------------
OutOfMemoryError (array clone):               OK
java.lang.OutOfMemoryError
   at [B.clone (Native Method)
   at extest.main (extest.java:360)


---------- no OK beyond this point --------------
NullPointerException (without catch):
Exception in thread "main" java.lang.NullPointerException
   at extest.main (extest.java:374)
